### PR TITLE
feat: Improve DX when developing against another repo

### PR DESCRIPTION
**Pull request by pawelgrimm**
## Short description

We are implementing a yalc-based workflow for working on Reactist and seeing those changed reflected in a locally running app that has a dependency on Reactist.

This partially addresses https://github.com/Doist/reactist/issues/675:
> To make it easier to work with Reactist locally, in development mode, alongside our product apps

## PR Checklist

-   [ ] ~~Added tests for bugs / new features~~
-   [x] Updated docs (storybooks, readme)
-   [ ] ~~Executed `npm run validate` and made sure no errors / warnings were shown~~
-   [ ] ~~Described changes in `CHANGELOG.md`~~
-   [ ] ~~Bumped version in `package.json` and `package-lock.json` (`npm --no-git-tag-version version <major|minor|patch>`) [ref](https://docs.npmjs.com/cli/v6/commands/npm-version)~~
-   [ ] ~~Reviewed and approved Chromatic visual regression tests in CI~~

## Versioning

This is a DX-only change that does not require a new version release.

### CODE REVIEW

**Review by craigcarlyle — APPROVED** [review]
🎉

File: README.md [hunk 1]
```diff
@@ -88,45 +88,65 @@ This boots up a development server with hot reloading on http://localhost:6006.
...
-# delete current reactist dependency
-rm -rf ./node_modules/@doist/reactist
-
-# link local reactist version
-npm link ../reactist
+yalc add @doist/reactist
 ```
 
-The relative path to reactist may need to be changed to match your local environment.
+In your real application you need to first delete the current _@doist/reactist_ dependency and then link to your local one.
```
Comment (gnapse) [inline README.md:112]: What's "your real application" vs. "your host application"?

File: README.md [hunk 2]
```diff
@@ -88,45 +88,65 @@ This boots up a development server with hot reloading on http://localhost:6006.
...
-rm -rf ./node_modules/@doist/reactist
+# restore the original reactist version
+yalc remove @doist/reactist
+# re-install reactist from npm
+npm install
+```
 
-# re-install reactist from npm (-E avoids updating the version / package-lock.json)
-npm install -E @doist/reactist
+For convenience, you can add a `dev:reactist` script in your host application to automate the process of adding and removing the local Reactist version:
```
Comment (gnapse) [inline README.md:124]: We'll be doing that in todoist-web, I assume. Right?

File: README.md [hunk 3]
```diff
@@ -88,45 +88,65 @@ This boots up a development server with hot reloading on http://localhost:6006.
...
+
+
 ## Development tips and tricks
 
 Independent of the development you operate in to produce a new build (e.g. before submitting a PR) run:
 
 ```sh
 npm run build
-```
+````
```
Comment (gnapse) [inline README.md:149]: There are problems with the backticks here that make the markdown rendering incorrect:

CleanShot 2024-07-23 at 21 27 20@2x

**Review by gnapse — COMMENTED** [review]
This is great! Can't wait to see it in action.

Unfortunately, I saw this today too late to try it. Will try this Wednesday. I did leave some comments here of a couple of things that are unclear to me.

**Pull request merged**